### PR TITLE
7 post comments to articles

### PR DESCRIPTION
**Pull request by GowerRob**
Hello,
Please RE-review my
Task 7 Posting comments by article id

~~Note... I'd appreciate some feedback on my understanding of the difference between a 400 and 404... I don't think I've fully comprehended it :) Thanks~~

### CODE REVIEW

File: __tests__/index.test.js [hunk 1]
```diff
@@ -129,6 +129,53 @@ describe('GET /api/articles',()=>{
...
+});
+
+
+describe('POST /api/articles/:article_id/comments',()=>{
+    test('get a 201 code and the inserted comment when posting with a valid username and article id',()=>{
+        const newComment=
+        {body:"This is a really good article", 
+        username:"lurker"};
+        
+        return request(app)
```
Comment (pippaamy) [inline __tests__/index.test.js:207]: can we have a 201 one test that ignores unnecessary properties

File: __tests__/index.test.js [hunk 2]
```diff
@@ -129,6 +129,53 @@ describe('GET /api/articles',()=>{
...
+
+    test('get a 400 code and error message when posting to a valid article but as an invalid username',()=>{
+        const newComment=
+        {body:"This is a really good article", 
+        username:"glasses2020"};
+        
+        return request(app)
+        .post('/api/articles/1/comments')
+        .send(newComment)
+        .expect(400)
```
Comment (pippaamy) [inline __tests__/index.test.js]: username not valid would be a 404

File: __tests__/index.test.js [hunk 3]
```diff
@@ -129,6 +129,53 @@ describe('GET /api/articles',()=>{
...
+    })
 
+    test('get a 400 code and error message when posting to an article that does not exist',()=>{
+        const newComment=
+        {body:"This is a really good article", 
+        username:"lurker"};
+        
+        return request(app)
+        .post('/api/articles/77/comments')
+        .send(newComment)
```
Comment (pippaamy) [inline __tests__/index.test.js:239]: what about if the article didn't exist and was "cheese"

File: __tests__/index.test.js [hunk 4]
```diff
@@ -129,6 +129,53 @@ describe('GET /api/articles',()=>{
...
+        .expect(400)
+        .then((response)=>{
+            expect(response.body.msg).toBe('Bad request')
+        })
+    })
+
+    test('get a 400 code and error message when posting to a valid article but as an invalid username',()=>{
```
Comment (pippaamy) [inline __tests__/index.test.js]: what if I posted a comment and left had missing fields

File: endpoints.json [hunk 5]
```diff
@@ -56,6 +56,30 @@
       }
         ]
     }
+  },
+  "POST /api/articles/:article_id/comments":{
+    "description": "post a comment to comments given a body, username and article_id in the endpoint",
+    "queries": [],
+    "requestBodyFormat":{
+      "body":"This is a really good article", 
+      "username":"lurker"
```
Comment (pippaamy) [inline endpoints.json:92]: nice!

**Review by pippaamy — COMMENTED** [review]
great job, just some tests to add! could you add those and pop in another pr :)

File: __tests__/index.test.js [hunk 6]
```diff
@@ -133,6 +133,15 @@ describe('GET /api/articles',()=>{
 
 
 describe('POST /api/articles/:article_id/comments',()=>{
+    test('get a 201 code',()=>{
+        const newComment={body:"This is a really good article", username:"lurker"};
+        
+        return request(app)
```
Comment (Jodiech) [inline __tests__/index.test.js:179]: Please add in another test for a status 201 - ignores unnecessary properties 👍

**Review by Jodiech — CHANGES_REQUESTED** [review]
Hey 👋 Great work on the amendments, I'd like you to add in an additional test for a 201 - ignores unnecessary properties. Let me know if you need any help with this 👍

**Review by pippaamy — COMMENTED** [review]
we're still missing a 400 for if there's missing required field(s), e.g. no username or body properties. can we add this test please?

**Review by pippaamy — COMMENTED** [review]
great, good to merge!

**Pull request merged**